src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -149,146 +167,238 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             }
         }
 
-
-        protected void WriteFullObjectJson(JsonWriter writer, object value, JsonSerializer serializer)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void WriteFullObjectJson(JsonWriter writer, object value, JsonSerializer serializer)

```

Comment:
going from `protected` to `private` will be a breaking change (custom classes extending `ResourceObjectConverter` could be calling it). 

I'm not too worried about changing it because its probably not used much, but its worth being aware of it